apps/crn-server/src/data-providers/contentful/interest-groups.data-provider.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  FetchGroupOptions,
+  GroupDataObject,
+  GroupTools,
+  GroupRole,
+  GroupLeader,
+  ListGroupDataObject,
+} from '@asap-hub/model';
+import {
+  GraphQLClient,
+  FETCH_INTEREST_GROUPS,
+  FETCH_INTEREST_GROUP_BY_ID,
+  FETCH_INTEREST_GROUPS_BY_USER_ID,
+  FetchInterestGroupByIdQuery,
+  FetchInterestGroupByIdQueryVariables,
+  FetchInterestGroupsQuery,
+  FetchInterestGroupsQueryVariables,
+  FetchInterestGroupsByUserIdQuery,
+  FetchInterestGroupsByUserIdQueryVariables,
+  InterestGroupsOrder,
+  InterestGroupsFilter,
+  Teams,
+} from '@asap-hub/contentful';
+
+import { InterestGroupDataProvider } from '../types';
+import {
+  parseContentfulGraphqlCalendarToResponse,
+  parseInterestGroupLeader,
+} from '../../entities';
+import { parseContentfulGraphQlUsers } from './users.data-provider';
+import { parseContentfulGraphQlTeams } from './teams.data-provider';
+
+type InterestGroupItem = NonNullable<
+  NonNullable<
+    FetchInterestGroupsQuery['interestGroupsCollection']
+  >['items'][number]
+>;
+
+type InterestGroupLeader = NonNullable<
+  NonNullable<
+    FetchInterestGroupsByUserIdQuery['interestGroupLeadersCollection']
+  >['items'][number]
+>;
+
+type InterestGroupsQueryResult =
+  FetchInterestGroupsQuery['interestGroupsCollection'];
+
+export class InterestGroupContentfulDataProvider
+  implements InterestGroupDataProvider
+{
+  constructor(private contentfulClient: GraphQLClient) {}
+
+  async fetchById(id: string): Promise<GroupDataObject | null> {
+    const { interestGroups } = await this.contentfulClient.request<
+      FetchInterestGroupByIdQuery,
+      FetchInterestGroupByIdQueryVariables
+    >(FETCH_INTEREST_GROUP_BY_ID, { id });
+
+    if (!interestGroups) {
+      return null;
+    }
+
+    return parseGraphQLInterestGroup(interestGroups);
+  }
+
+  private async fetchByUserId(
+    userId: string,
+    options: FetchGroupOptions,
+  ): Promise<ListGroupDataObject> {
+    const { take = 10, skip = 0 } = options;
```

Comment:
I've noticed that the `take` default in the fetch method is `20`. I wonder if we could leave this as 20 as well since the place where fetchByUserId is not doing paginated fetches, maybe would be better to leave `take` as the max we can without having the query complexity error